packages/swapper/src/swappers/JupiterSwapper/endpoints.ts

```diff
@@ -59,34 +71,13 @@ export const jupiterApi: SwapperApi = {
 
     if (!isExecutableTradeStep(step)) throw Error('Unable to execute step')
 
-    const solanaInstructions = step.jupiterTransactionMetadata?.instructions?.map(instruction =>
-      adapter.convertInstruction(instruction),
-    )
-
-    if (!isSolanaFeeData(step.feeData.chainSpecific)) throw Error('Unable to execute step')
-
-    const buildSwapTxInput: BuildSendApiTxInput<KnownChainIds.SolanaMainnet> = {
+    const getFeeDataInput: GetFeeDataInput<KnownChainIds.SolanaMainnet> = {
       to: '',
-      from,
       value: '0',
-      accountNumber: step.accountNumber,
-      chainSpecific: {
-        addressLookupTableAccounts: step.jupiterTransactionMetadata?.addressLookupTableAddresses,
-        instructions: solanaInstructions,
-        computeUnitLimit: step.feeData.chainSpecific?.computeUnits,
-        computeUnitPrice: step.feeData.chainSpecific?.priorityFee,
-      },
-    }
-
-    const { txToSign } = await adapter.buildSendApiTransaction(buildSwapTxInput)
-
-    const getFeeDataInput: GetFeeDataInput<KnownChainIds.SolanaMainnet> = {
-      to: txToSign.to,
-      value: txToSign.value,
       chainSpecific: {
         from,
         addressLookupTableAccounts: step.jupiterTransactionMetadata?.addressLookupTableAddresses,
-        instructions: txToSign.instructions as TransactionInstruction[] | undefined,
```

Comment:
This was the reason why it was failing, txToSign.instructions was not `TransactionInstruction` but `SolanaTxInstruction`

Also obviously why would we need to build the whole TX to estimate fees... my bad!